addon/models/resource.js

```diff
@@ -388,12 +388,12 @@ const RelatedProxyUtil = Ember.Object.extend({
     @param {Ember.ObjectProxy|Ember.ArrayProxy} proxyFactory
     @return {PromiseProxy} proxy
   */
-  createProxy: function (resource, proxyFactory) {
+  createProxy: function (resource, type, proxyFactory) {
     const relation = this.get('relationship');
     const url = this.proxyUrl(resource, relation);
-    const service = resource.container.lookup('service:' + pluralize(relation));
+    const service = resource.container.lookup('service:' + pluralize(type));
     let promise = this.promiseFromCache(resource, relation, service);
-    promise = promise || service.findRelated(relation, url);
+    promise = promise || service.findRelated(type, url);
```

Comment:
I think this is already the case: the type defines the service to use in line
```
const service = resource.container.lookup('service:' + pluralize(type));
```
Resource defines the url to use:
```
const relation = this.get('relationship');
const url = this.proxyUrl(resource, relation);
```
So the `url` is created based on the `this.get('relationship')`, which gets the value of `resource`.
The `type` is passed in and is either passed to the relation (e.g. `hasOne`) in a hash or takes the same value as `resource`

...

The only problem I have now is that I probably should have passed the type like the relationship is passed: in the create, not as a parameter
```
const util = RelatedProxyUtil.create({'relationship': relationName});
```
would become
```
const util = RelatedProxyUtil.create({'relationship': relationName, 'type': type});
```
and would probably be more clear...